packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP connection, channel, server, and handler"""
+
+import asyncio
+import json
+import logging
+import threading
+from asyncio import CancelledError
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from typing import Optional, Set, cast
+from urllib.parse import parse_qs, urlparse
+from uuid import uuid4
+
+import yaml
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext, URI
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPChannel:
+    """A wrapper for an RESTful API with an internal HTTPServer."""
+
+    def __init__(
+        self,
+        address: Address,
+        host: str,
+        port: int,
+        api_spec: str,
+        connection_id: PublicId,
+        excluded_protocols: Optional[Set[PublicId]] = None,
+    ):
+        """
+        Initialize a channel.
+
+        :param address: the address
+        """
+        self.address = address
+        self.host = host
+        self.port = port
+        self.connection_id = connection_id
+        self.in_queue = None  # type: Optional[asyncio.Queue]
+        self.loop = None  # type: Optional[asyncio.AbstractEventLoop]
+        self.excluded_protocols = excluded_protocols
+        # self.thread = Thread(target=self.receiving_loop)
+        self.lock = threading.Lock()
+        self.stopped = True
+
+        if api_spec is not None:
+            self._process_api(api_spec)
+        logger.info(f"Initialized the HTTP Server on HOST: {host} and PORT: {port}")
+
+    def _process_api(self, api_spec):
+        """Process the api_spec.
+
+        Decode all paths to get envelops each, and put it in the agent's inbox.
+        """
+        try:
+            paths = api_spec["paths"]
+        except KeyError:
+            logger.error("Key 'paths' not found in API spec.")
+
+        for path_name, path_value in paths.items():
+            try:
+                envelope = self._decode_path(path_name, path_value)
+                assert self.in_queue is not None, "Input queue not initialized."
+                assert self.loop is not None, "Loop not initialized."
+                # Ok to put these Envelopes in_queue during HTTPChannel initialization?
+                asyncio.run_coroutine_threadsafe(self.in_queue.put(envelope), self.loop)
+            except ValueError:
+                logger.error(f"Bad formatted path: {path_name}")
+            except Exception as e:
+                logger.error(f"Error when processing a path. Message: {str(e)}")
+
+    def _decode_path(self, path_name: str, path_value: dict, separator: str = "/"):
```

Comment:
There is no point to this. Unnecessary overhead as you can just feed the specification file to the skill directly. I'd remove this method.

Reply:
Deleted, the skill will just receive the path of the specification file.